### PR TITLE
Automatically find the center and the rings

### PR DESCRIPTION
**Pull request by sameera2004**

### CODE REVIEW

**Comment by sameera2004** [comment]
Example notebook added https://github.com/scikit-beam/scikit-beam-examples/pull/65

File: skbeam/core/roi.py [hunk 1]
```diff
@@ -691,3 +693,54 @@ def lines(end_points, shape):
         label += 1
         label_array[rr, cc] = label
     return label_array
+
+
+def auto_find_center_rings(avg_img, sigma=1, no_rings=4, min_samples=3,
+                           residual_threshold=1):
+    """This will find the center of the speckle pattern and the radii of the
+    most intense rings
```
Comment (licode) [inline skbeam/core/roi.py]: Please add more doc like this is based on ransac algorithm. Are you going to add other algorithms? Or ransac is the most standard in the field?

File: skbeam/core/roi.py [hunk 2]
```diff
@@ -691,3 +693,54 @@ def lines(end_points, shape):
         label += 1
         label_array[rr, cc] = label
     return label_array
+
+
+def auto_find_center_rings(avg_img, sigma=1, no_rings=4, min_samples=3,
+                           residual_threshold=1):
```
Comment (licode) [inline skbeam/core/roi.py]: it is better to add max_trials as an input to this function, so we can control from outside.

Comment (sameera2004) [inline skbeam/core/roi.py]: Sure I will

File: skbeam/core/roi.py (hunk 1 above)
Comment (sameera2004) [inline skbeam/core/roi.py]: @licode I will add more doc to this function

**Comment by sameera2004** [comment]
@licode thanks for the suggestions,  fixed the documentation and added max_trails ready to review.

File: skbeam/core/roi.py [hunk 3]
```diff
@@ -691,3 +693,60 @@ def lines(end_points, shape):
...
+    avg_img : 2D array
+        shape of the image
+    sigma : float, optional
+        Standard deviation of the Gaussian filter.
+    no_rings : int, optional
+        number of rings
+    min_sample : int, optional
+        The minimum number of data points to fit a model to.
+    residual_threshold : float, optional
+        Maximum distance for a data point to be classified as an inlier.
```
Comment (licode) [inline skbeam/core/roi.py]: max_trials is missing here.

File: skbeam/core/roi.py [hunk 4]
```diff
@@ -691,3 +693,62 @@ def lines(end_points, shape):
...
+    Note
+    ----
+    scikit-image ransac method(http://www.imagexd.org/tutorial/lessons/1_ransac.html)
+    is used to automatically find the center and the most intense rings.
+    """
+
+    image = img_as_float(color.rgb2gray(avg_img))
+    edges = feature.canny(image, sigma)
+    coords = np.column_stack(np.nonzero(edges))
+    edge_pts_xy = coords[:, ::-1]
```
Comment (licode) [inline skbeam/core/roi.py]: any reason to reverse x and y here?

Comment (licode) [inline skbeam/core/roi.py]: OK, I guess x,y position needs to be consistent with CircleModel's input.

File: skbeam/core/tests/test_roi.py [hunk 5]
```diff
@@ -397,3 +397,14 @@ def test_lines():
...
+
+def test_auto_find_center_rings():
+
+    x = np.linspace(-5, 5, 200)
+    X, Y = np.meshgrid(x, x)
+    image = 100*np.cos(np.sqrt(x**2 + Y**2))**2 + 50
+
+    center, image, radii = roi.auto_find_center_rings(image, no_rings=2)
+
+    assert_equal((99, 99), center)
```
Comment (licode) [inline skbeam/core/tests/test_roi.py]: please also check radii, which is another main output from ransac.

Comment (sameera2004) [inline skbeam/core/tests/test_roi.py]: @licode fixed the documentation, added to check the output radii. ready for review.

**Comment by licode** [comment]
Thanks. It looks good to me. It will be much helpful if ipython notebook example can be added later.

**Comment by sameera2004** [comment]
@licode Thanks, please find the example notebook here https://github.com/scikit-beam/scikit-beam-examples/pull/65

**Pull request merged**